### ISSUE TITLE
remove visible browser blue from focus state - leave focus state, but not the browser blue

### ISSUE THREAD

**Issue by spacedino** [ampproject/ampstart#371]
![image](https://cloud.githubusercontent.com/assets/7339798/24729684/3b2b788c-1a14-11e7-86f6-8e3561323c7b.png)

**Comment by ericlindley-g** [comment]
@spacedino — could you elaborate? Is there another visual style you want to apply, or you want to remove the default browser visual affordance for focus entirely?

**Comment by spacedino** [comment]
@ericlindley-g  thats right, just remove the blue

**Comment by ericlindley-g** [comment]
Got it — the reason I ask is that, while removing just the visual affordance still supports users with limited vision, it stops supporting users with limited mobility/dexterity. The visual affordance for focus helps them navigate pages using interfaces that might just be a single button, because they need a clear visual indication of which parts of the page have focus at a given time. The best practice that I'm aware of is to keep the default browser blue, because that's what these users are used to. But I'd be curious if there are other good options.

**Comment by spacedino** [comment]
@ericlindley-g that totally makes sense for selections. this is a slightly different use case  - the button is treating a "pressed" indication as a "focus" but ultimately there is no need for the "focus" state in this situation. After the user presses the button the page changes, adding a new focus. The only technical reason I believe (I could be wrong) it stays is because a modal is launched on top of the page so it may be preserving where the user was. If this is the case we need to address what happens after the modal closes, because ultimately that button is only a trigger for the modal. It doesn't indicate a state change and therefore a visual indicator is not needed. It's an outlier, all of the other buttons on the page have the blue line removed for the same reason. Does that make sense? We can also discuss offline.

**Comment by ericlindley-g** [comment]
Discussed offline — sounds like there might be a feature request for AMPHTML core — "On tap, remove focus from amp-lightbox trigger and apply to lightboxed container"

@aghassemi — what do you think?

**Comment by aghassemi** [comment]
already tracked: https://github.com/ampproject/amphtml/issues/4892 
- lightbox container should get the focus for A11Y reasons anyway.
- on closing the lightbox however focus will be moved back to the opener.

**Comment by ericlindley-g** [comment]
Got it—thanks!  I'm assuming there's no workaround in the meantime (?). If not, then I think we can close this issue and wait for the fix in core.

**Comment by aghassemi** [comment]
Well, if the issue is about the ugliness of the blue outline, we should consider that same thing can happen outside of the open lightbox case (e.g. imagine a button that stays visible on the page when tapped but changes something else on the page like a client-side `search` button!). The right approach in my opinion is to provided a good on-focus look for ampstart buttons so we can remove the browser-provided outline without hurting low-vision users. /cc @spacedino

**Comment by ericlindley-g** [comment]
Gotcha—I'm not super-familiar with the space, but I feel like what I've read in terms of best practices is to just go with the browser default, because that's what users expect. Have you seen other good patterns for this?

**Comment by aghassemi** [comment]
I think we can provide a better design than the browser default. Material Design raised the button onfocus. Bootstrap provides a more subtle outline using box-shadow that matches the button color, etc...

**Comment by ericlindley-g** [comment]
Makes sense—still makes me nervous, though, in part because such a style as Material used couldn't be applied across the board to all elements (as opposed to outline styling). Though they might have research that backs up the decision. I'll send a note to them to see.

FYI: here's [one strongly-titled piece on the topic](https://www.tjvantoll.com/2013/01/28/stop-messing-with-the-browsers-default-focus-outline/)

**Comment by aghassemi** [comment]
another option is to keep the outline for `:focus` but remove it for `:active` meaning if users taps the button, no outline is displayed but when TABing through buttons, outline will be displayed.

**Comment by spacedino** [comment]
@aghassemi this is what i'm asking to do. i am confused though because i am NOT advocating to remove the outline for keyboard users. i just want it to match the behavior for the buttons on the rest of the site. we just forgot to override the <button> tag @camelburrito . @ericlindley-g we were right though about the issue @aghassemi linked to :)

**Comment by ericlindley-g** [comment]
Weird, yeah — I didn't notice that this was on the :active state, before. I'm not sure about best practices for that, but I think that's pretty typical to change (actually seems unusual to give it styling that looks like :focus). 

If we change this, we should change this for the component, as well, which is showing a blue outline for the :active state, as well (and doesn't seem to change color on :active state, as spec'd out in the mocks — though in the mocks the :active state is called "pressed" and the default button state is called "active". Might help to be rigorous about how we refer to these going forward)
